phdi/fhir/geospatial/core.py

```
@@ -72,3 +72,43 @@ def _store_lat_long_extension(address: dict, lat: float, long: float) -> None:
                 ],
             }
         )
+
+    @staticmethod
+    def _store_census_tract_extension(address: dict, census_tract: str) -> None:
+        """
+        Adds appropriate extension data for census tract for each element in an address
+        line, if the field isn't already present, to a given FHIR-formatted dictionary
+        holding address fields. Add the extension data directly to the input dictionary,
+        leaving census tract as a FHIR-identified geolocation element.
+
+        :param address: A FHIR formatted dictionary holding address fields
+        :param census_tract: The census tract to add to the FHIR data as an extension
+        """
+
+        # Append with a properly resolving URL for FHIR's canonical censusTract
+        # structure definition, as all extensions are required to have this
+        # attribute; see https://www.hl7.org/fhir/extensibility.html
+
+        census_extension = {
+            "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-censusTract",
+            "valueString": census_tract,
+        }
+
+        if address.get("_line") is None:
+            address["_line"] = []
+        for element_counter in range(len(address["line"])):
+            try:
+                if address["_line"][element_counter].get("extension", "") is not None:
```

I think this if statement is unnecessary.  The line below should raise the exceptions you are using to handle edge cases.

---

Ah good catch! I added the AttrubuteError but forgot to remove the if statement.